### PR TITLE
feat: supporing hybrid search group_by

### PR DESCRIPTION
**Pull request by MrPresent-Han**
related: #35096

### CODE REVIEW

File: internal/proxy/search_reduce_util.go [hunk 1]
```diff
@@ -428,23 +468,215 @@ func rankSearchResultData(ctx context.Context,
...
+func compareKey(keyI interface{}, keyJ interface{}) bool {
+	switch keyI.(type) {
+	case int64:
+		return keyI.(int64) < keyJ.(int64)
+	case string:
+		return keyI.(string) < keyJ.(string)
+	}
+	return false
+}
+
```
Comment (MrPresent-Han) [inline internal/proxy/search_reduce_util.go:490]: different score strategy is used here to prepare different scorer

File: internal/proxy/search_reduce_util.go [hunk 2]
```diff
@@ -455,22 +687,54 @@ func rankSearchResultData(ctx context.Context,
...
 			},
 		}
 	default:
-		return nil, errors.New("unsupported pk type")
+		return errors.New("unsupported pk type")
+	}
+	return nil
+}
+
+func rankSearchResultDataByPk(ctx context.Context,
```
Comment (MrPresent-Han) [inline internal/proxy/search_reduce_util.go:712]: for normal hybridSearch, this rank branch is executed

File: internal/querynodev2/delegator/delegator.go [hunk 3]
```diff
@@ -375,12 +381,7 @@ func (sd *shardDelegator) Search(ctx context.Context, req *querypb.SearchRequest
 			}
 			results[i] = result
 		}
-		var ret *internalpb.SearchResults
```
Comment (MrPresent-Han) [inline internal/querynodev2/delegator/delegator.go:378]: I removed useless reduce here, and return directly for reduction outside in searchChannel

File: internal/querynodev2/services.go [hunk 4]
```diff
@@ -753,69 +753,41 @@ func (node *QueryNode) Search(ctx context.Context, req *querypb.SearchRequest) (
...
-	tr.RecordSpan()
-	var result *internalpb.SearchResults
-	var err2 error
-	if req.GetReq().GetIsAdvanced() {
-		result, err2 = segments.ReduceAdvancedSearchResults(ctx, toReduceResults, req.Req.GetNq())
-	} else {
-		result, err2 = segments.ReduceSearchResults(ctx, toReduceResults, segments.NewReduceInfo(req.Req.GetNq(),
-			req.Req.GetTopk(),
-			req.Req.GetExtraSearchParam().GetGroupByFieldId(),
-			req.Req.GetExtraSearchParam().GetGroupSize(),
```
Comment (MrPresent-Han) [inline internal/querynodev2/services.go:796]: currently in milvus query framework, only one channel will be searched, queried in one request, which is triggered by proxy, So I simplify code here and remove unnessary reduction between different channels which will never happen at all

**Comment by MrPresent-Han** [comment]
rerun ut

File: internal/querynodev2/segments/result.go [hunk 5]
```diff
@@ -129,53 +137,14 @@ func ReduceAdvancedSearchResults(ctx context.Context, results []*internalpb.Sear
 		IsAdvanced: true,
 	}
 
-	for _, result := range results {
```
Comment (czs007) [inline internal/querynodev2/segments/result.go:132]: why remove this?

File: internal/util/reduce/reduce_info.go [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+type ResultInfo struct {
+	nq             int64
+	topK           int64
+	metricType     string
+	pkType         schemapb.DataType
+	offset         int64
+	groupByFieldId int64
+	groupSize      int64
+	isAdvance      bool
+}
```
Comment (czs007) [inline internal/util/reduce/reduce_info.go:16]: add motify options

Comment (MrPresent-Han) [reply internal/util/reduce/reduce_info.go:16]: changed

File: internal/proxy/task_search.go [hunk 7]
```diff
@@ -403,6 +404,11 @@ func (t *searchTask) initAdvancedSearchRequest(ctx context.Context) error {
 			zap.Int64s("plan.OutputFieldIds", plan.GetOutputFieldIds()),
 			zap.Stringer("plan", plan)) // may be very large if large term passed.
 	}
+	if len(t.queryInfos) > 0 {
```
Comment (czs007) [inline internal/proxy/task_search.go:407]: to do parse from rankParams

**Pull request merged**